src/ontodia/data/sparql/sparqlDataProvider.ts

```diff
@@ -132,11 +135,24 @@ export class SparqlDataProvider implements DataProvider {
     }
 
     elementInfo(params: { elementIds: string[]; }): Promise<Dictionary<ElementModel>> {
-        const ids = params.elementIds.map(escapeIri).map(id => ` (${id})`).join(' ');
+        const blankIds: string[] = [];
+
+        const elementIds = params.elementIds.filter(id => !BlankNodes.isEncodedBlank(id));
+        const additionalData: SparqlResponse<ElementBinding> = (
+            this.options.acceptBlankNodes ?
+            BlankNodes.elementInfo(params.elementIds) :
+            null
+        );
+
+        if (elementIds.length === 0 && this.options.acceptBlankNodes) {
+            return Promise.resolve(getElementsInfo(additionalData, params.elementIds));
+        }
+
+        const ids = elementIds.map(escapeIri).map(id => ` (${id})`).join(' ');
         const {defaultPrefix, dataLabelProperty, elementInfoQuery} = this.settings;
         const query = defaultPrefix + resolveTemplate(elementInfoQuery, {ids, dataLabelProperty});
 
-        return this.executeSparqlQuery<ElementBinding>(query)
+        return this.executeSparqlQuery<ElementBinding>(query, {additionalData})
```

Comment:
It's better to create function like `concat(firstData, secondData)` and chain another `.then()` call instead of adding `{additionalData}` param to `executeSparqlQuery`:
```
return this.executeSparqlQuery<ElementBinding>(query)
    .then(data => concat(data, additionalData))
    .then(...)
```